.github/workflows/sonar.yml

```diff
@@ -0,0 +1,47 @@
+name: Run sonar analysis
+
+on:
+  push:
+    branches: [ master ]
+
+env:
+  SONAR_URL: https://sonarcloud.io
+  SONAR_ORGANIZATION: telecominfraproject
+  SONAR_PROJECT_KEY: wlan-cloud-opensync-protobuf
+
+jobs:
+  sonar:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Set up JDK 1.12
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.12
+    - name: Cache Protobuf Binary
+      uses: actions/cache@v1
+      with:
+        path: ~/protobuf-2.6.1-bin
+        key: ${{ runner.os }}-protobuf
+    - name: Check if protobuf compiler binary is present
+      run: ls /home/runner/protobuf-2.6.1-bin/bin/protoc
+    - name: Build protobuf binary if needed
+      if: failure() # previous step failed - protoc file is not present
+      run: |
+        curl -L -O https://github.com/google/protobuf/releases/download/v2.6.1/protobuf-2.6.1.tar.gz
+        tar xf protobuf-2.6.1.tar.gz
+        cd protobuf-2.6.1
+        LDFLAGS="-static-libgcc -static-libstdc++" ./configure --disable-shared --prefix=/home/runner/protobuf-2.6.1-bin
+        cd src
+        make protoc
+        make install
+    - name: Configure ~/.m2/toolchains.xml file
+      if: always()
+      run: |
+        cp toolchains.xml ~/.m2/toolchains.xml
+    - name: Run sonar
+      run: |
```

Comment:
You may want to add      " if: always()" in here, like in the other steps. What happens is we don't want to re-build protobuf compiler binary every time, so on line 29 we perform a check if that binary is installed. When that check fails the protobuf binary is built in the next step on line 31. The side-effect of that check is that it can mark the step as failed, and unless subsequent steps have "if: failure()" or "if: always()" they will be skipped.

Reply:
good catch, thanks!

Reply:
@dmitry-toptygin-connectus-ai refactored, merged check and install steps to make workflow cleaner a bit